qualtop/wizard/account_invoice_refund.py

```diff
@@ -0,0 +1,26 @@
+from odoo import models, api
+
+
+class AccountInvoiceRefund(models.TransientModel):
+    _inherit = "account.invoice.refund"
+
+    @api.multi
+    def invoice_refund(self):
+        res = super(AccountInvoiceRefund, self).invoice_refund()
+        data_refund = self.read(['filter_refund'])[0]['filter_refund']
+        if data_refund == 'refund' and res.get('domain', False):
+            """ To obtain the credit note id is necessary search in
+            res['domain'] which has the form
+            [('type', '=', 'out_refund'), ('id', 'in', [5])]. The credit note
+            id is in res['domain'][1][2][0]
+            """
```

Comment:
This is a little bit less hardwired, what do you think?
`
cn_id = [x[2][0] for x in res['domain'] if x[0] == 'id']
if not id:
   return res
cn_id = cn_id[0]

`

Reply:
Change executed